store/src/java/com/zimbra/cs/service/mail/ParseMimeMessage.java

```diff
@@ -912,6 +912,7 @@ private static void attachPart(MimeMultipart mmp, ItemId iid, String part, Strin
         if (!iid.isLocal()) {
             Map<String, String> params = new HashMap<String, String>(3);
             params.put(UserServlet.QP_PART, part);
+            params.put(UserServlet.QP_DISP, getDisposition(contentDisposition));
```

Comment:
getDisposition() looks good to encapsulate the check for `Part.ATTACHMENT.equals(contentDisposition) ? "a" : "i"` but its just a single check. I think we can remove the method and directly use the conditional check here itself.